sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusConnectionStringBuilder.cs

```diff
@@ -0,0 +1,449 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.ServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Text;
+    using Primitives;
+
+    /// <summary>
+    /// Used to generate Service Bus connection strings.
+    /// </summary>
+    internal class ServiceBusConnectionStringBuilder
```

Comment:
This class is not one that I'd recommend keeping around long term.   It lacks the single responsibility described by its name and, instead, was given all sorts of tangentially related operations.   The functionality is largely covered in many of the Event Hubs classes that were already adopted.

Reply:
Will create an issue for connection string parsing.

Reply:
Please note that this class is helpful when constructing a connection from various parts that are not necessarily available as constructor parameters on all the clients. If you plan to remove it (https://github.com/Azure/azure-sdk-for-net/issues/9884), ensure it's possible to construct the clients with all the options passed in.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/9884